### PR TITLE
JAMES-3310 Make DeletedMessageVaultHook failure sensitive

### PR DESCRIPTION
**Pull request by blackheaven**

### CODE REVIEW

**Comment by blackheaven** [comment]
We'll see them burn

File: mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/DeletedMessageVaultHook.java [hunk 1]
```diff
@@ -135,6 +135,7 @@ public final int hashCode() {
         return mapperFactory.getMessageIdMapper(session)
             .findReactive(ImmutableList.of(deletedMessageMailboxContext.getMessageId()), MessageMapper.FetchType.Full)
             .next()
+            .switchIfEmpty(Mono.error(new RuntimeException("Cannont find " + deletedMessageMailboxContext.getMessageId())))
```
Comment (chibenwa) [inline mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/DeletedMessageVaultHook.java:138]: :+1: the problem is here, but I did not understood why

Comment (blackheaven) [reply mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/DeletedMessageVaultHook.java:138]: Me neither, it seems we are not able to read written data.

Comment (remk) [reply mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/DeletedMessageVaultHook.java:138]: Cassandra nconsistency on one node :trollface:  ?

**Comment by blackheaven** [comment]
no log

test this please

**Comment by blackheaven** [comment]
test this please

**Comment by blackheaven** [comment]
```
[a10ed47b34c079cd04048c31487e5c285c283fdb] [ERROR]   RabbitMQEventDeadLettersIntegrationTest.groupIdOfFailedEventShouldBeStoredInDeadLetter:322 1 expectation failed.
[a10ed47b34c079cd04048c31487e5c285c283fdb] JSON path . doesn't match.
[a10ed47b34c079cd04048c31487e5c285c283fdb] Expected: iterable over ["org.apache.james.webadmin.integration.rabbitmq.RabbitMQEventDeadLettersIntegrationTest$RetryEventsListenerGroup"] in any order
[a10ed47b34c079cd04048c31487e5c285c283fdb]   Actual: []
```

test this please

**Comment by chibenwa** [comment]
```
[a10ed47b34c079cd04048c31487e5c285c283fdb] Running: Verify if file exists in workspace
[a10ed47b34c079cd04048c31487e5c285c283fdb] Running: Shell Script
```

test this please

**Comment by blackheaven** [comment]
test this please

**Comment by blackheaven** [comment]
test this please

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**